projects/hipsparse/CMakeLists.txt

```
@@ -1,289 +1,278 @@
-  LDCONFIG
-  LDCONFIG_DIR ${HIPSPARSE_CONFIG_DIR})
-
-# Build docs
-if(BUILD_DOCS)
-add_subdirectory(docs)
-endif()
+    NAME ${package_name}
+    DESCRIPTION "ROCm SPARSE library"
+    MAINTAINER "hipSPARSE Maintainer <hipsparse-maintainer@amd.com>"
+    LDCONFIG
+    LDCONFIG_DIR ${HIPSPARSE_CONFIG_DIR}
+)
 
-#
-# ADDITIONAL TARGETS FOR CODE COVERAGE
-#
-if(BUILD_CODE_COVERAGE)
-  #
-  # > make coverage_cleanup (clean coverage related files.)
-  # > make coverage GTEST_FILTER=<>
-  # will run:
-  #  > make coverage_analysis GTEST_FILTER=<> (analyze tests)
-  #  > make coverage_output (generate html documentation)
-  #
-
-  #
-  # Run coverage analysis
-  #
-  add_custom_target(coverage_analysis
-    COMMAND echo Coverage GTEST_FILTER=\${GTEST_FILTER}
-    COMMAND ./clients/staging/hipsparse-test --gtest_filter=\"\${GTEST_FILTER}\"
-    WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
-    )
+# --- Documentation ---
 
-  add_dependencies(coverage_analysis hipsparse)
+if(HIPSPARSE_BUILD_DOCS)
+    add_subdirectory(docs)
+endif()
 
-  #
-  # Prepare coverage output
-  # This little script is generated because the option '--gcov-tool <program name>' of lcov cannot take arguments.
-  #
-  add_custom_target(coverage_output
-    DEPENDS coverage_analysis
-    COMMAND mkdir -p lcoverage
-    )
+# --- Code coverage targets ---
 
-  #
-  # Generate coverage output.
-  #
-  add_custom_command(TARGET coverage_output
-    COMMAND lcov --directory . --base-directory . --capture -o lcoverage/raw_main_coverage.info
-    COMMAND lcov --remove lcoverage/raw_main_coverage.info "'/opt/*'" "'/usr/*'" -o lcoverage/main_coverage.info
-    COMMAND genhtml lcoverage/main_coverage.info --output-directory lcoverage
+if(HIPSPARSE_ENABLE_COVERAGE)
+    add_custom_target(coverage_cleanup
+        COMMAND find ${CMAKE_BINARY_DIR} -name *.gcda -delete
+        WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
+        COMMENT "Cleaning up coverage data files"
     )
-
-  add_custom_target(coverage DEPENDS coverage_output)
-
-  #
-  # Coverage cleanup
-  #
-  add_custom_target(coverage_cleanup
-    COMMAND find ${CMAKE_BINARY_DIR} -name *.gcda -delete
-    WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
+    
+    add_custom_target(coverage_analysis
+        COMMAND echo Coverage GTEST_FILTER=\${GTEST_FILTER}
+        COMMAND ./clients/staging/hipsparse-test --gtest_filter=\"\${GTEST_FILTER}\"
+        WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
+        COMMENT "Running tests for coverage analysis"
+    )
+    
+    add_dependencies(coverage_analysis hipsparse)
+    
+    add_custom_target(coverage_output
+        DEPENDS coverage_analysis
+        COMMAND mkdir -p lcoverage
+        COMMAND lcov --directory . --base-directory . --capture -o lcoverage/raw_main_coverage.info
+        COMMAND lcov --remove lcoverage/raw_main_coverage.info "'/opt/*'" "'/usr/*'" -o lcoverage/main_coverage.info
+        COMMAND genhtml lcoverage/main_coverage.info --output-directory lcoverage
```

```suggestion
        DEPENDS coverage_analysis
        COMMAND mkdir -p lcoverage
        COMMAND lcov --directory . --base-directory . --capture -o "${hipsparse_coverage_dir}/raw_main_coverage.info"
        COMMAND lcov --remove "${hipsparse_coverage_dir}/raw_main_coverage.info" "'/opt/*'" "'/usr/*'" -o lcoverage/main_coverage.info
        COMMAND genhtml "${hipsparse_coverage_dir}/main_coverage.info" --output-directory lcoverage
```

This will ensure that these paths don't get out of sync.